### PR TITLE
pdfjs-dist: Update some definitions to match implementation in v2.1.266

### PR DESCRIPTION
**Pull request by echo99**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/mozilla/pdf.js/blob/v2.1.266/src/display/api.js>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/pdfjs-dist/index.d.ts [hunk 1]
```diff
@@ -260,58 +261,128 @@ interface PDFDocumentProxy {
...
 
     /**
      * TODO: return type of Promise<???>
      *  A promise that is resolved with a lookup table for mapping named destinations to reference numbers.
      **/
-    getDestinations(): PDFPromise<any[]>;
+    getDestinations(): Promise<any[]>;
+
+    /**
+     * TODO: return type of Promise<???>
```
Comment (armanio123) [inline types/pdfjs-dist/index.d.ts]: Are these TODOs still valid?

**Review by armanio123 — COMMENTED** [review]
I don't have much context on the library, so there's just a few comments. The rest LGTM.

**Review by andrewbranch — CHANGES_REQUESTED** [review]
Requesting changes to address @armanio123’s comments so the bot knows what’s going on.

File: types/pdfjs-dist/index.d.ts [hunk 2]
```diff
@@ -248,6 +240,15 @@ interface PDFProgressData {
     total: number;
 }
 
+interface PDFDocumentStats {
+    streamTypes: object; // Used stream types in the document (an item is set to true if specific stream ID was used in the document).
```
[the remark this answers is not in the document]
Comment (echo99) [reply types/pdfjs-dist/index.d.ts]: Sure! I was following the existing comment styles in the file so wasn't sure which was preferred, but improving intellisense is definitely a good reason to use jsdoc style comments.

File: types/pdfjs-dist/index.d.ts (hunk 1 above)
Comment (echo99) [reply types/pdfjs-dist/index.d.ts]: Yeah. Unfortunately the documentation in PDF.js doesn't specify clearly what these methods return other than a promise, so I was unable to determine what the real type should be

Comment (andrewbranch) [reply types/pdfjs-dist/index.d.ts]: I guess the TODO should at least be taken out of the JSDoc comment so users don’t see it

**Review by andrewbranch — CHANGES_REQUESTED** [review]
Double check the version that you’re pulling these from—a lot of the methods I spot-checked, like `getViewerOptions` and `getPageLayout`, seem to have been added in 2.2, not 2.1. You just need to update the version string in the header accordingly to 2.2 or 2.3 (if you choose 2.2, please check that there are no 2.3 methods added here—I didn’t check exhaustively). Thanks!

**Pull request closed without merging**